app/controllers/application_controller.rb

```diff
@@ -5,4 +5,8 @@ class ApplicationController < ActionController::Base
   def current_user
    @current_user ||= User.find(session[:user_id]) if session[:user_id]
   end
+
+  def render_404
```

Comment:
that's some fine inheritance and abstraction(?) you got there! (trying to practice using the 4 pillars in sentences as often as possible lol)

Reply:
approve of both this addition and the follow up commentary on this addition

Reply:
Dare I say... Polymorphism?

Reply:
Looks good